x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.test.ts

```diff
@@ -21,18 +21,115 @@ expect.addSnapshotSerializer({
   },
 });
 const files = glob.sync(path.join(__dirname, '/tests/*.yml'));
-let allFields: Field[] = [];
+let fields: Fields = [];
 for (const file of files) {
   const fieldsYML = readFileSync(file, 'utf-8');
-  allFields = allFields.concat(safeLoad(fieldsYML));
+  fields = fields.concat(safeLoad(fieldsYML));
 }
 
 test('flattenFields function recursively flattens nested fields and renames name property with path', () => {
-  const flattened = flattenFields(allFields);
+  const flattened = flattenFields(fields);
   expect(flattened).toMatchSnapshot('flattenFields');
 });
 
 test('dedupFields function remove duplicated fields when parsing multiple files', () => {
-  const deduped = dedupFields(allFields);
+  const deduped = dedupFields(fields);
   expect(deduped).toMatchSnapshot('dedupFields');
 });
+
+test('transformField maps field types to kibana index pattern data types', () => {
+  expect(transformField({ name: 'testField' }).type).toBe('string');
+  expect(transformField({ name: 'testField', type: 'half_float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'scaled_float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'integer' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'long' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'short' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'byte' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'keyword' }).type).toBe('string');
+  expect(transformField({ name: 'testField', type: 'string' }).type).toBe(undefined);
```

Comment:
Looking at https://github.com/elastic/beats/blob/master/libbeat/kibana/fields_transformer.go#L292 , and without further investigating, I would have expected this to be the other way round, i.e. that fields with no type end up as string fields.

What is the story behind this?

Reply:
It's a bit confusing and I can't answer as to why, only that the tests say it should be that way, I believe: https://github.com/elastic/beats/blob/master/libbeat/kibana/fields_transformer_test.go#L205

Reply:
The first test does expect a field with no `type` property to be a string field OR if type is an empty string.  But this test here says if  a type has "string" for a value that it should be undefined: https://github.com/elastic/beats/blob/master/libbeat/kibana/fields_transformer_test.go#L205.  This could be a bug in the Go code, but I went off the tests.

Reply:
> Not sure why "string" would be passed in as a type but i think it's meant to represent an invalid type

Assuming that the field types that go in are elasticsearch field types, that seems to be the case: `string` doesn't exist, only `text` and `keyword` (cf. https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html) Thanks for the clarification!

Reply:
I'm going to update the test so its easier to understand.